### PR TITLE
add addon test for tkgs cluster creation

### LINKED ISSUE #2449: TC01: Able to create/delete CC Guest cluster on supervisor and verify if all addons are successfully deployed
Area: Addons deployment (Guest cluster)

### PR DESCRIPTION
**Pull request by ggpaue**
### What this PR does / why we need it
Adds logic to verify cluster addons reconciliation success in integration tests.

### Which issue(s) this PR fixes

Fixes #2449 

### Release note

```release-note
Automate testing to check if addons are successfully deployed when guest cluster got created
```

### Additional information
Adds logic to verify class cluster upgrade for integration tests.

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/shared/package.go [hunk 1]
```diff
@@ -620,3 +623,33 @@ func GeneratePackageSecretName(clusterName, carvelPkgRefName string) string {
...
+	wlcClient, _, _, _, err := getClients(ctx, tempFilePath)
+	Expect(err).NotTo(HaveOccurred())
+
+	By(fmt.Sprintf("Verify addon packages on workload cluster %q matches clusterBootstrap info on supervisor cluster %q", clusterName, svClusterName))
+	err = checkClusterCB(ctx, mngclient, wlcClient, svClusterName, constants.TkgNamespace, clusterName, namespace, InfrastructureName, false)
+	Expect(err).To(BeNil())
+
+	By(fmt.Sprintf("Verify addon packages on workload cluster %q matches clusterBootstrap info on supervisor cluster %q", clusterName, svClusterName))
+	err = checkClusterCB(ctx, mngclient, wlcClient, svClusterName, constants.TkgNamespace, clusterName, namespace, InfrastructureName, false)
+	Expect(err).To(BeNil())
```
Comment (maralavi) [inline pkg/v1/tkg/test/tkgctl/shared/package.go:652]: Why this block is repeated?

File: pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go [hunk 2]
```diff
@@ -52,7 +61,8 @@ var _ = Describe("TKGS ClusterClass based workload cluster tests", func() {
 			err = tkgctlClient.DeleteCluster(deleteClusterOptions)
 		})
 
-		It("should successfully create a cluster", func() {
+		It("should successfully create a cluster and verify addons information", func() {
```
Comment (maralavi) [inline pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go:64]: ```addons information```-->```successful addons reconciliation``` (please change other occurrences too)

File: pkg/v1/tkg/test/tkgctl/shared/package.go (hunk 1 above)
Comment (ggpaue) [reply pkg/v1/tkg/test/tkgctl/shared/package.go]: deleted the extra block

File: pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go (hunk 2 above)
Comment (ggpaue) [reply pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go]: updated test description

File: pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go [hunk 3]
```diff
@@ -25,17 +29,22 @@ const (
...
+		w             *os.File
+		ctx           context.Context
+		svClusterName string
 	)
 	JustBeforeEach(func() {
 		err = tkgctlClient.CreateCluster(clusterOptions)
 	})
 
 	BeforeEach(func() {
+		ctx = context.TODO()
```
Comment (adduarte) [inline pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go:43]: nit.. you might want to use context.BACKGROUND()

**Comment by adduarte** [comment]
lgtm

**Review by haoranleo — APPROVED** [review]
Looks good! (Just test to see if I can approve

Seems my approval is not valid XD

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Review by maralavi — APPROVED** [review]

**Pull request merged**